l10n_ch_pain_credit_transfer/models/account_payment_order.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# copyright 2018 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class AccountPaymentOrder(models.Model):
+    _inherit = 'account.payment.order'
+
+    @api.model
+    def _get_line_key(self, line):
+        return super(AccountPaymentOrder, self)._get_line_key(line) + (
+            line.currency_id.name,
+        )
+
+    @api.model
+    def generate_start_payment_info_block(
+            self, parent_node, payment_info_ident,
+            priority, local_instrument, category_purpose, sequence_type,
+            requested_date, eval_ctx, gen_args):
+        return super(AccountPaymentOrder,
+                     self).generate_start_payment_info_block(
+            parent_node, "self.name + '-' "
```

Comment:
maybe better to computer 
```
 payment_info_ident = "self.name + '-' "
                         "+ requested_date.replace('-', '') + '-' "
                         "+ priority + '-' "
                         "+ optional_args[0] + '-' "
                         "+ local_instrument + '-' "
                         "+ category_purpose"
```
outside call, to increase readability?

Reply:
I agree it would be cleaner to use a variable to avoid this strange indent as @oleksandrpaziuk mentionned.

@oleksandrpaziuk that code is just missing the wrapping '()'

Reply:
@yvaucher sorry I use code only for example purpose )
next time put the valid code into a comment )

Reply:
Done